src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,1241 @@
+    std::unique_ptr<DocGenerator> _filterTemplate;
+    metrics::Operation _operation;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: insertMany
+ *      OperationCommand:
+ *        Documents:
+ *        - { a : 1 }
+ *        - { b : 1 }
+ */
+struct InsertManyOperation : public BaseOperation {
+
+    InsertManyOperation(YAML::Node opNode,
+                        bool onSession,
+                        mongocxx::collection collection,
+                        genny::DefaultRandom& rng,
+                        metrics::Operation operation)
+        : _onSession{onSession}, _collection{collection}, _operation{operation} {
+        auto documents = opNode["Documents"];
+        if (!documents && !documents.IsSequence()) {
+            throw InvalidConfigurationException(
+                "'insertMany' expects a 'Documents' field of sequence type.");
+        }
+        for (auto&& document : documents) {
+            _docTemplates.push_back(std::move(value_generators::makeDoc(document, rng)));
+        }
+        // TODO: parse insert options.
+    }
+
+    void run(mongocxx::client_session& session) override {
+        for (auto&& docTemplate : _docTemplates) {
+            bsoncxx::builder::stream::document doc{};
+            docTemplate->view(doc);
+            _writeOps.push_back(doc.extract());
+        }
+        auto ctx = _operation.start();
+        (_onSession) ? _collection.insert_many(session, _writeOps, _options)
+                     : _collection.insert_many(_writeOps, _options);
+        ctx.success();
+    }
+
+private:
+    mongocxx::collection _collection;
+    const bool _onSession;
+    std::vector<bsoncxx::document::value> _writeOps;
+    mongocxx::options::insert _options;
+    metrics::Operation _operation;
+    std::vector<std::unique_ptr<DocGenerator>> _docTemplates;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: startTransaction
+ *      OperationCommand:
+ *        Options:
+ *          WriteConcern:
+ *            Level: majority
+ *            Journal: true
+ *          ReadConcern:
+ *            Level: snapshot
+ *          ReadPreference:
+ *            ReadMode: primaryPreferred
+ *            MaxStalenessSeconds: 1000
+ */
+
+struct StartTransactionOperation : public BaseOperation {
+
+    StartTransactionOperation(YAML::Node opNode,
+                              bool onSession,
+                              mongocxx::collection collection,
+                              genny::DefaultRandom& rng,
+                              metrics::Operation operation) {
+        if (!opNode.IsMap())
+            return;
+        if (opNode["Options"]) {
+            _options = opNode["Options"].as<mongocxx::options::transaction>();
+        }
+    }
+
+    void run(mongocxx::client_session& session) override {
+        try {
+            session.start_transaction(_options);
+        } catch (mongocxx::operation_exception& exception) {
+            throw;
+        }
```

Comment:
no need to catch&rethrow

Reply:
In addition, could you add a TODO or add to the TODO ticket to wrap all the run methods with a try/catch and rethrow as genny::MongoException. Not particularly useful here but should come in handy for CRUD operations